cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala

```diff
@@ -48,11 +48,8 @@ final class ClusterBootstrap(implicit system: ExtendedActorSystem) extends Exten
         discovery
 
       case otherDiscoveryMechanism ⇒
-        val implClazz = system.settings.config.getString(otherDiscoveryMechanism + ".class")
-        log.info("Bootstrap using [{}] discovery mechanism, instantiating [{}]", otherDiscoveryMechanism, implClazz)
-        system.dynamicAccess
-          .createInstanceFor[SimpleServiceDiscovery](implClazz, List(classOf[ActorSystem] → system))
-          .get
+        log.info("Bootstrap using `akka.discovery` mechanism: {}", otherDiscoveryMechanism)
+        ServiceDiscovery(system).loadServiceDiscovery(otherDiscoveryMechanism)
```

Comment:
This now also 'loads' the discovery mechanism - does this mean it becomes available when using akka-discovery 'directly' as well? Would that lead to confusion? Or was that already the case?

Reply:
It won't be used by the globally configured discovery but will already be cached if another area calls loadServiceDiscovery. So I don't think users will see any change.